enumerable.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+# rubocop:disable all
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield self[i]
+        i += 1
+      end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+    i = 0
+      while i < size
+        yield(self[i], i)
+        i += 1
+      end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+    array2 = []
+      i = 0
+      while i < size
+        array2 << self[i] if yield self[i]
+        i += 1
+      end
+    array2
+  end
+
+  def my_all?(pattern = nil)
+    test = true
+    if(block_given?)
+      self.my_each do |x|
+        test = test && yield(x)
+      end 
+    else
+      if(pattern.nil?)
+        self.my_each do |x|
+          test = test && (x==true)
+        end
+      else
+        self.my_each do |x|
+          test = test && (x.match?(pattern))
+        end
+      end
+    end
+    return test
+  end
+
+  def my_any?(pattern = nil)
+    test = false
+    if(block_given?)
+      self.my_each do |x|
+        test = test || yield(x)
+      end
+    else
+      if(pattern.nil?)
+        self.my_each do |x|
+          test = test || (x == true)
+        end
+      else
+        self.my_each do |x|
+          test = test || (x.match?(pattern))
+        end
+      end
+    end
+    return test
+  end
+
+  def my_none?(pattern = nil)
+    test = true
+    if (block_given?)
```

Comment:
```
%w{ant bear cat}.none? { |word| word.length == 5 } #=> true
%w{ant bear cat}.none? { |word| word.length >= 4 } #=> false
%w{ant bear cat}.none?(/d/)                        #=> true
[1, 3.14, 42].none?(Float)                         #=> false
[].none?                                           #=> true
[nil].none?                                        #=> true
[nil, false].none?                                 #=> true
[nil, false, true].none?                           #=> false
```

- [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern